### PR TITLE
Add fix for CIESM pr

### LINKED ISSUE #1340: Dataset problem: CMIP6 pr_Amon_CIESM_historical_r1i1p1f1 appears to have the wrong units
**Describe the dataset issue**
A clear and concise description of what the problem with the data is is. Here are some guidelines that you can use
right out the box by filling in the blanks (note that, if needed we will assist you with writing a fix; also note that
if the problem is with a CMIP6 dataset, we will alert the ESGF and/or the model developers for them to fix the problem
in the long run; fixes for CMIP5 by the model developers are no longer possible):

- Data file has been changed by you in any way (if you answer yes the issue will be void and closed, we are not
supporting custom-made datasets that present problems, it is your resposability to fix those problems):
- Project (CMIP5/CMIP6/OBS/obs4MIPs/etc): CMIP6
- Full dataset description (fill out as many as you know, please):
  - dataset: CIESM
  - experiment: historical
  - mip: Amon
  - grid: gr
  - version: v20200417
  - variable used: pr
- Problems encountered (please describe what the actual problems are: e.g. wrong standard name, issue with dimensions):
The mean value of the file CMIP6/CMIP/THU/CIESM/historical/r1i1p1f1/Amon/pr/gr/v20200417/pr_Amon_CIESM_historical_r1i1p1f1_gr_185001-201412.nc
is 2.8436496e-08 kg m-2 s-1, while typical values are a factor 1000 bigger.
```python
>>> import iris
>>> c = iris.load_cube("/home/bandela/climate_data/CMIP6/CMIP/THU/CIESM/historical/r1i1p1f1/Amon/pr/gr/v20200417/pr_Amon_CIESM_historical_r1i1p1f1_gr_185001-201412.nc")
/home/bandela/conda/envs/esmvaltool/lib/python3.9/site-packages/iris/fileformats/cf.py:861: UserWarning: Missing CF-netCDF measure variable 'areacella', referenced by netCDF variable 'pr'
  warnings.warn(
>>> c.units
Unit('kg m-2 s-1')
>>> c.core_data().mean().compute()
2.8436496e-08
```

- Pointer to existing copy of data on ESGF node (it would be very useful if you could provide a physical
fullpath to the file(s) that are causing the problem, e.g. on CEDA Jasmin or DKRZ):
CMIP6/CMIP/THU/CIESM/historical/r1i1p1f1/Amon/pr/gr/v20200417/pr_Amon_CIESM_historical_r1i1p1f1_gr_185001-201412.nc

- Other notes and mentions: I suspect the units are wrong and should be 'Mg m-2 s-1' instead of 'kg m-2 s-1'

**Assign the right people**
If you are already a member of the ESMValTool GitHub project, please assign Valeriu Predoi (valeriupredoi) and
Klaus Zimmermann (zklaus) to the issue. They will then check the issue raised and propagate it further to the
data model developers.

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Closes #1340

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/cmor/_fixes/cmip6/ciesm.py [hunk 1]
```diff
@@ -19,9 +19,20 @@ def fix_data(self, cube):
...
+
+class Pr(Fix):
+    """Fixes for pr."""
+
+    def fix_metadata(self, cube):
+        """Fix metadata.
+
+        The values of v20200417 are off by a factor 1000.
+        """
+        cube.data = cube.core_data() * 1000.
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cmip6/ciesm.py:37]: this is `fix_data()` man, and the args of the baseclass function are `(self, cubes)` hence the Codacy issue

**Review by valeriupredoi — CHANGES_REQUESTED** [review]

File: esmvalcore/cmor/_fixes/cmip6/ciesm.py [hunk 2]
```diff
@@ -19,9 +19,20 @@ def fix_data(self, cube):
...
         metadata = cube.metadata
         cube *= 100
         cube.metadata = metadata
         return cube
+
+
+class Pr(Fix):
+    """Fixes for pr."""
+
+    def fix_metadata(self, cube):
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cmip6/ciesm.py:32]:
```suggestion
    def fix_data(self, cube):
```

File: tests/integration/cmor/_fixes/cmip6/test_ciesm.py [hunk 3]
```diff
@@ -35,3 +35,18 @@ def test_cl_fix_data(cl_cube):
     fix = Cl(None)
     out_cube = fix.fix_data(cl_cube)
     assert out_cube.data == [100.0]
+
+
+def test_pr_fix():
+    """Test `Pr.fix_metadata`."""
```
Comment (valeriupredoi) [inline tests/integration/cmor/_fixes/cmip6/test_ciesm.py:41]:
```suggestion
    """Test `Pr.fix_data`."""
```

File: tests/integration/cmor/_fixes/cmip6/test_ciesm.py [hunk 4]
```diff
@@ -35,3 +35,18 @@ def test_cl_fix_data(cl_cube):
...
+    cube = iris.cube.Cube(
+        [2.82e-08],
+        var_name='pr',
+        units='kg m-2 s-1',
+    )
+
+    fix = Fix.get_fixes('CMIP6', 'CIESM', 'Amon', 'pr')[0]
+
+    out_cube = fix.fix_metadata(cube)
```
Comment (valeriupredoi) [inline tests/integration/cmor/_fixes/cmip6/test_ciesm.py:50]:
```suggestion
    out_cube = fix.fix_data(cube)
```

**Comment by zklaus** [comment]
Thanks for taking this on @bouweandela. This fix will do the wrong thing once the data is corrected. There is already an erratum about this at https://errata.es-doc.org/static/view.html?uid=d4a11704-808a-5869-b94d-234170eb70ea, so I wonder if we can/should take that into account?

**Comment by valeriupredoi** [comment]
ah hawkeye @zklaus :eagle: But given that was created in January, and still no correct data, I think we should just go ahead with the fix. We should, however, maybe add a catchment see if the data has indeed changed? Like a `min()` on the current data

**Comment by zklaus** [comment]
My thoughts exactly. The tricky bit is determining whether it is applicable or not. We could of course look at statistics (max might be better than min; perhaps mean is even better), but the problem with that is that it needs to realize the data, which we would of course like to avoid. I have no good answer and will even accept moving forward just like this. Perhaps we should add a link to the erratum in any case as a comment.

**Comment by valeriupredoi** [comment]
yeah I realized about data realization (pun intended :grin: ) right after I posted the comment - maybe we can take a slice of the cube first? Sure thing, you mind posting a nudge on the erratum page, I feel you are more famous than me there :grin:

**Comment by bouweandela** [comment]
I could realize a bit of the data, say the first 10 timesteps, and see if the mean is larger than 10^-5? That is quite fast.

**Comment by zklaus** [comment]
Sounds good. I was just wondering: At the fix we don't have access to the version, do we?

**Comment by bouweandela** [comment]
Not guaranteed, because it is not stored in the file, but only in the path and we have no clue how users organize their data.

**Comment by zklaus** [comment]
Alright. Let's do statistics over a limited subset then.

**Comment by valeriupredoi** [comment]
> Not guaranteed, because it is not stored in the file, but only in the path and we have no clue how users organize their data.

it should be stored in the cube metadata if they do it correctly, no?

**Comment by zklaus** [comment]
> > Not guaranteed, because it is not stored in the file, but only in the path and we have no clue how users organize their data.
> 
> it should be stored in the cube metadata if they do it correctly, no?

No. CMIP6 decided deliberately not to have the version as an attribute. This is to allow for the easy replacement with a new version.

**Review by zklaus — APPROVED** [review]

**Comment by bouweandela** [comment]
> CMIP7 will exclude the model name too? 😆

And the variable name and units, that will save a lot of data downloading because then you can reuse the same file for every analysis 🤣

**Comment by zklaus** [comment]
Excellent ideas, you two. You can always get the missing metadata by querying a dozen web services in the cloud.

**Comment by valeriupredoi** [comment]
I reckon we need to get proactive in the CMIP7 effort - we'll be what carbon fiber was to motorsports - light and sturdy - ok just light, in our case :laughing:

**Review by valeriupredoi — APPROVED** [review]

**Pull request merged**